lzw.py

```diff
@@ -85,18 +76,60 @@ def fill_zero_bytes(compress_data):
     :param: compress_data:
     :return: compress_data
     """
-    if zero_fill := len(compress_data) % 8:
-        compress_data = convert_int_to_bits(0, zero_fill) + compress_data
+    if zero_fill := len(compress_data) % BYTE_LEN:
+        compress_data = convert_int_to_bits(0, BYTE_LEN - zero_fill) + compress_data
     return compress_data
 
 
+def bitstring_to_bytes(bitstr):
+    hex_str = binascii.hexlify(int(bitstr, 2).to_bytes((len(bitstr) + 7) // BYTE_LEN, 'big')).decode()
+    return bytes.fromhex(hex_str)
+
+
+def get_decode_element(stream, reading_size) -> int:
+    if stream.pos - reading_size < 0:
+        reading_size = stream.pos
+
+    stream.pos -= reading_size
+
+    value: int = stream.read(f'uint{reading_size}')
+    stream.pos -= reading_size
+    return value
+
+
+def index_to_binary(element, writing_size):
+    return bytes(''.join([bin(int(val))[2:].zfill(writing_size) for val in element.split(',')]), 'utf-8')
+
+
+def fill_zero_hexa(hexa_data, binary_data_len):
+
+    while len(hexa_data[2:]) < binary_data_len / 4:
+        hexa_data = '0x0' + hexa_data[2:]
+    return hexa_data
+
+
+def update_code_size_decode(table_size, code_size):
+    if table_size == int(math.pow(2, code_size)) and code_size < MAX_WRITING_SIZE:
+        return code_size + 1
+    return code_size
+
+
+def get_first_element(concats_colors):
+    comma_index = concats_colors.find(",")
+    if comma_index != -1:
+        result = concats_colors[:comma_index]
+    else:
+        result = concats_colors
+    return result
+
+
 def encode(uncompressed_data, color_table_size):
```

Comment:
doc about this and the decoder and their flow here or maybe better in the format descriptor files